site/en/guides/thread-primer/router-selection.md

```diff
@@ -175,9 +175,126 @@ What you've learned:
     maintain the CDS
 *   The MLE Link Request process is used to establish Router-Router links
 
+## Check your understanding
+
+<div>
+  <devsite-multiple-choice>
+    <div>Which of these rules are not enforced by a Connected Dominating Set
+    (CDS)?</div>
+    <div>
+      <div>There is a Router-only path between any two Routers.</div>
+      <div>Incorrect.</div>
+    </div>
+    <div>
+      <div>Any one Router in a Thread network can reach any other Router by
+        staying entirely within the set of Routers.</div>
+      <div>Incorrect.</div>
+    </div>
+    <div>
+      <div>Every End Device in a Thread network is directly connected to a
+        Router.</div>
+      <div>Incorrect.</div>
+    </div>
+    <div correct>
+      <div>Only one Router in a Thread network may be a Border Router.</div>
+      <div>Correct. A Thread network may have multiple Border Routers.</div>
+    </div>
+  </devsite-multiple-choice>
+</div>
+
+<div>
+  <devsite-multiple-choice>
+    <div>Why might a Router be removed from a Thread network?</div>
+    <div correct>
+      <div>To reduce the Routing state below the maximum of 32 Routers.</div>
+      <div>Correct. Thread networks strive to maintain an optimal number of
+        Routers. The most Routers that any Thread network should have is 32.
+      </div>
+    </div>
+    <div>
+      <div>To free up channels.</div>
+      <div>Incorrect. The number of routers has no relation to channel usage
+      or capacity.</div>
+    </div>
+    <div correct>
+      <div>To allow the election of new Routers in other parts of the network
+      when needed.</div>
+      <div>Correct. Reducing the number of active Routers in one part of a
+      Thread network increases its ability to ramp up routing capacity
+      elsewhere.</div>
+    </div>
+  </devsite-multiple-choice>
+</div>
+
+<div>
+  <devsite-multiple-choice>
+    <div>Which of the following statements correctly describe what happens when
+      a Child device upgrades to a Router?</div>
+    <div correct>
+      <div>The device remains on the network with the same address.</div>
+      <div>That's right. The device stays connected to the network. Only its
+        role changes.</div>
+    </div>
+    <div>
+      <div>The device must initiate the MLE Attach process to establish a new
+        network connection.</div>
+      <div>Incorrect. The MLE Attach process is used when a device joins
+        a Thread network. The process followed by a device upgrading to a Router
+        is called the MLE Link Request Process.</div>
+    </div>
+    <div>
+      <div>The other Routers 'vote' to decide whether to allow the device to
+        become a Router.</div>
+      <div>Incorrect. The Leader decides whether to allow the device to become a
+      Router, based on the distributed algorithm that maintains the CDS.</div>
```

Comment:
The Leader does not actually have enough knowledge to maintain the CDS. However, the Leader does enforce some rules (e.g. no more than 32 routers, unique Router ID for each router).